src/frontend/src/flows/addDevice/manage/verifyTentativeDevice.ts

```diff
@@ -48,116 +115,126 @@ const pageContent = (alias: string) => {
   });
 };
 
+export const verifyTentativeDevicePage = (
+  props: Parameters<typeof verifyTentativeDeviceTemplate>[0],
+  container?: HTMLElement
+): void => {
+  const contain =
+    container ?? (document.getElementById("pageContent") as HTMLElement);
+  render(verifyTentativeDeviceTemplate(props), contain);
+};
+
 /**
  * Page to verify the tentative device: the device verification code can be entered and is the checked on the canister.
- * @param userNumber anchor of the authenticated user
- * @param tentativeDevice the tentative device to be verified
- * @param endTimestamp timestamp when the registration mode expires
  * @param connection authenticated II connection
+ * @param alias of the tentative device to be verified
+ * @param endTimestamp timestamp when the registration mode expires
  */
-export const verifyDevice = async (
-  userNumber: bigint,
-  connection: AuthenticatedConnection,
-  tentativeDevice: DeviceData,
-  endTimestamp: bigint
-): Promise<void> => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(tentativeDevice.alias), container);
-  init(userNumber, connection, endTimestamp);
-};
+export const verifyTentativeDevice = async ({
+  connection,
+  alias,
+  endTimestamp,
+}: {
+  connection: AuthenticatedConnection;
+  alias: string;
+  endTimestamp: bigint;
+}): Promise<void> => {
+  const countdown = AsyncCountdown.fromNanos(endTimestamp);
 
-const init = (
-  userNumber: bigint,
-  connection: AuthenticatedConnection,
-  endTimestamp: bigint
-) => {
-  const countdown = setupCountdown(
-    endTimestamp,
-    document.getElementById("timer") as HTMLElement,
-    async () => {
-      await displayError({
-        title: "Timeout Reached",
-        message:
-          'The timeout has been reached. For security reasons the "add device" process has been aborted.',
-        primaryButton: "Ok",
-      });
-      await renderManage(userNumber, connection);
-    }
+  // The verification page
+  const verification = new Promise<VerifyResult | { canceled: null }>(
+    (resolve) =>
+      verifyTentativeDevicePage({
+        alias,
+        cancel: async () => {
+          await withLoader(() => connection.exitDeviceRegistrationMode());
+          // NOTE: we resolve before stopping the countdown to avoid Promise.race below
+          // thinking it was a "stop-by-timeout"
+          resolve({ canceled: null });
```

Comment:
nth: would not mind the `.stop()` before the `resolve` for readability reason - i.e. when I read `resolve`, even though I know it isn't the end of the function, I kind of except spontaneously that it's its end

Reply:
Unfortunately can't do that, see comment above! Or we'd need some other way of differentiating timeout vs cancel